src/xsarsea/cross_spectra_core.py

```diff
@@ -0,0 +1,307 @@
+"""
+author Antoine Grouazel:
+original code provided by Frederic Nouguier the 29 March 2022
+"""
+import numpy as np
+import xarray as xr
+import scipy
+import matplotlib.pyplot as plt
+import xrft
+import logging
+import xsar
+import os
+
+def get_imagette_indice(onetiff,wv_slc_meta):
+    good_indice = None
+    # find the indice of the tiff
+    imagette_number = os.path.basename(onetiff).split('-')[-1].replace('.tiff','')
+    logging.info('imagette_number : %s',imagette_number)
+    for ddi,ddname in enumerate(wv_slc_meta.subdatasets) :
+        if 'WV_'+imagette_number in ddname :
+            logging.info("matching ddname : %s",ddname)
+            good_indice = ddi
+    logging.info('indice in meta sentinel1 gdal driver is : %s',good_indice)
+    return good_indice
+
+def read_slc(onetiff,slice_subdomain=None,resolution=None,resampling=None):
```

Comment:
Perhaps it's possible to use a `Sentinel1Dataset` object instead of `onetiff` ? This will allow:
  - suppress `get_imagette_indice`
  - remove `import xsar` ( xsar is currently not an xsarsea dep)

Reply:
`xsar` import removed.